pkgs/by-name/si/sigil/package.nix

```diff
@@ -30,37 +26,50 @@ stdenv.mkDerivation rec {
   nativeBuildInputs = [
     cmake
     pkg-config
-    makeWrapper
+    qt6.wrapQtAppsHook
   ];
 
   buildInputs = [
     boost
     xercesc
-    qtbase
-    qttools
-    qtwebengine
-    qtsvg
+    qt6.qtbase
+    qt6.qttools
+    qt6.qtwebengine
+    qt6.qtsvg
     python3Packages.lxml
   ];
 
   prePatch = ''
     sed -i '/^QTLIB_DIR=/ d' src/Resource_Files/bash/sigil-sh_install
   '';
 
-  dontWrapQtApps = true;
+  installPhase = lib.optionalString stdenv.hostPlatform.isDarwin ''
+    runHook preInstall
 
-  preFixup = ''
-    wrapProgram "$out/bin/sigil" \
-       --prefix PYTHONPATH : $PYTHONPATH \
-       ''${qtWrapperArgs[@]}
+    mkdir -p $out/Applications
+    mv bin/Sigil.app $out/Applications
+    # https://github.com/NixOS/nixpkgs/issues/186653
+    chmod -x $out/Applications/Sigil.app/Contents/lib/*.dylib
+
+    runHook postInstall
   '';
 
+  dontWrapQtApps = stdenv.hostPlatform.isLinux;
```

Comment:
What is wrong with wrapping on linux?

Reply:
$out/bin/sigil is a shell script, so it will not be wrapped automatically. [^1]

But this line just prevents a no-op, maybe it should be removed.  What do you think?

[^1]: https://nixos.org/manual/nixpkgs/unstable/#qt-runtime-dependencies

Reply:
Maybe documentation is outdated, but it properly wrapped for me automatically.

Reply:
Just tried. You are right.